enumarable.rb

```diff
@@ -0,0 +1,220 @@
+    if self.is_a? Array
+      self.my_each do |element|
+        case yield element
+          when true then return false
+        end
+      end
+    elsif self.is_a? Hash
+      self.my_each do |key, element|
+        case yield key, element
+          when true then return false
+        end
+      end
+    else
+      return "You can't use my_none? method on #{self.class} class"
+    end
+    true
+  end
+  
+  def my_count(num = nil)
+    counter=0
+    if num == nil
+    if self.is_a?(Array) && block_given?
+      self.my_each do |element|
+        case yield element
+          when true then counter += 1
+        end
+      end
+    elsif self.is_a?(Hash) && block_given?
+      self.my_each do |key, element|
+        case yield key, element
+          when true then counter += 1
+        end
+      end
+    elsif self.is_a?(Array) && !block_given?
+      return self.length
+    elsif self.is_a?(Hash) && !block_given?
+      return self.keys.length
+    else
+      return "You can't use my_count? method on #{self.class} class"
+    end
+  else
+    if self.is_a? Array
+      self.my_each do |element|
+        case num == element
+          when true then counter += 1
+        end
+      end
+    elsif self.is_a? Hash
+      self.my_each do |key, element|
+        case element == num 
+          when true then counter += 1
+        end
+      end
+    else
+      return "You can't use my_count? method on #{self.class} class"
+    end
+  end
+    counter
+  end
+  
+  def my_map(proc="default")
+    result = []
+    if self.is_a? Array
+      if proc.is_a? Proc
+        self.my_each do |element|
+          result << proc.call(element)
+        end
+      elsif proc == "default"
+        self.my_each do |element|
+          result << yield(element)
+        end
+      else
+        return "U need to pass a proc"
+      end
+
+    elsif self.is_a? Hash
+      if proc.is_a? Proc
+        self.my_each do |key,element|
+          result << proc.call(key, element)
+        end
+      elsif proc == "default"
+        self.my_each do |key, element|
+          result << yield(key, element)
+        end
+      else
+        return "U need to pass a proc"
+      end
+    else
+      return "You can't use my_map method on #{self.class} class"
+    end
+    result
+  end
+  
+  def my_inject(acc="undefined")
+    temp = self
+    if self.is_a? Array
+      if acc == "undefined"
+        acc = self[0]
+        temp = self[1..-1]
+      end
+      temp.my_each do |element|
+        acc = yield(acc, element)
+      end
+    elsif self.is_a? Hash
+      if acc == "undefined"
+        acc = self[self.keys[0]]
+        temp.delete(temp.keys[0])
+      end
+      temp.my_each do |key, element|
+        acc = yield(acc, key, element)
+      end
+    else
+      return "You can't use my_inject method on #{self.class} class"
+    end
+    acc
+  end
+  
+    def multiply_els
+      self.my_inject {|acc, num| acc*num}
+    end
+  
+  end
+
+  include Enumerable
```

Comment:
`include Enumerable`not needed here, remove next 2 lines as well

Reply:
Thanks Inhaq for your efforts